gateway-ha/src/main/java/io/trino/gateway/ha/config/DataStoreConfiguration.java

```diff
@@ -101,43 +104,55 @@ public void setRunMigrationsEnabled(boolean runMigrationsEnabled)
         this.runMigrationsEnabled = runMigrationsEnabled;
     }
 
-    public MysqlConfiguration getMysqlConfiguration()
+    public MySqlConfiguration getMySqlConfiguration()
     {
-        return mysqlConfiguration;
+        return mySqlConfiguration;
     }
 
-    public void setMysqlConfiguration(MysqlConfiguration mysqlConfig)
+    public void setMysqlConfiguration(MySqlConfiguration mysqlConfig)
     {
-        this.mysqlConfiguration = mysqlConfig;
+        this.mySqlConfiguration = mysqlConfig;
     }
 
-    public DataStoreBackend getDataStoreBackendType()
+    public DataStoreType getDataStoreType()
     {
-        if (dataStoreBackendType != null) {
-            return dataStoreBackendType;
+        if (dataStoreType != null) {
+            return dataStoreType;
+        }
+        if (jdbcUrl == null) {
+            throw new IllegalStateException("jdbcUrl must be set");
         }
 
-        if (jdbcUrl != null) {
-            String url = jdbcUrl.toLowerCase(Locale.ROOT);
-            if (url.startsWith("jdbc:oracle:")) {
-                return DataStoreBackend.ORACLE;
-            }
-            if (url.startsWith("jdbc:mysql:")) {
-                return DataStoreBackend.MYSQL;
-            }
-            if (url.startsWith("jdbc:postgresql:")) {
-                return DataStoreBackend.POSTGRES;
-            }
-            if (url.startsWith("jdbc:h2:")) {
-                return DataStoreBackend.H2;
+        try {
+            Enumeration<Driver> drivers = DriverManager.getDrivers();
+            while (drivers.hasMoreElements()) {
+                Driver driver = drivers.nextElement();
+                if (driver.acceptsURL(jdbcUrl)) {
+                    String driverClass = driver.getClass().getName();
+                    if (driverClass.contains("oracle")) {
+                        return DataStoreType.ORACLE;
+                    }
```

Comment:
This fuzzy string match against class name doesn't seem right to me. We have all of the driver classes on the classpath, can we do an actual match against the class name? Like adding a field to `DataStoreType` which tracks the expected instance of `Class<?>` for that `DataStoreType`.  Then this iteration can become something like:
```
while (drivers.hasMoreElements()) {
                Driver driver = drivers.nextElement();
                if (driver.acceptsURL(jdbcUrl)) {
                    for (DataStoreType dataStoreType : DataStoreType.values() {
                      if (dataStoreType.driverClass() == driver.getClass()) {
                        return dataStoreType;
                      }
                    }
                    break;
                 }
}
```